packages/paste-core/components/badge/src/hooks.tsx

```diff
@@ -0,0 +1,99 @@
+import * as React from 'react';
+import {Button} from '@twilio-paste/button';
+import {safelySpreadBoxProps} from '@twilio-paste/box';
+import {useUIDSeed} from '@twilio-paste/uid-library';
+
+import type {BadgeProps, BadgeChildren} from './types';
+import {hasValidAnchorVariantProps, hasValidButtonVariantProps, safelySpreadProps} from './utils';
+
+const BUTTON_DENY_LIST = ['fullWidth', 'as', 'size', 'type', 'variant', 'href'];
+const ANCHOR_DENY_LIST = ['variant', 'as', 'onHover'];
+
+const DEFAULT_ICON_SIZE = 'sizeIcon10';
+const FOCUSABLE_STYLES = {
+  textDecoration: 'underline',
+  cursor: 'pointer',
+  _hover: {textDecoration: 'none'},
+  _focus: {boxShadow: 'shadowFocus', textDecoration: 'none'},
+};
+
+export type Wrapper = React.FC<{children: React.ReactNode}>;
+export type StyleProps =
+  | {
+      textDecoration: 'underline';
+      cursor: 'pointer';
+      _hover: {textDecoration: 'none'};
+      _focus: {boxShadow: 'shadowFocus'; textDecoration: 'none'};
+    }
+  | Record<string, never>;
+
+export const useFocusableVariants = (
+  props: Omit<BadgeProps, 'variant' | 'children'>
+): {
+  wrapper: Wrapper;
+  styleProps: StyleProps;
+  spanProps: Partial<Omit<BadgeProps, 'variant' | 'children'>> | Omit<BadgeProps, 'variant' | 'children'>;
+} => {
+  const isValidButtonVariant = hasValidButtonVariantProps(props);
+  const isValidAnchorVariant = hasValidAnchorVariantProps(props);
+
+  const styleProps = isValidAnchorVariant || isValidButtonVariant ? {...FOCUSABLE_STYLES} : {};
+
+  const redactedProps = safelySpreadBoxProps(props);
+
+  const buttonProps = safelySpreadProps(redactedProps, BUTTON_DENY_LIST);
+  const anchorProps = safelySpreadProps(redactedProps, ANCHOR_DENY_LIST);
+
+  if (isValidButtonVariant) {
+    return {
+      styleProps,
+      wrapper: ({children}) => (
+        <Button variant="reset" size="reset" type="button" {...buttonProps}>
+          {children}
+        </Button>
+      ),
+      spanProps: {},
+    };
+  }
+  if (isValidAnchorVariant) {
+    return {
+      styleProps,
+      wrapper: ({children}) => (
+        <Button variant="reset" size="reset" type="button" as="a" {...anchorProps}>
+          {children}
+        </Button>
+      ),
+      spanProps: {},
+    };
+  }
+
+  return {
+    styleProps,
+    wrapper: ({children}) => <>{children}</>,
+    spanProps: redactedProps,
+  };
+};
+
+export const useResizeChildIcons = (children: BadgeChildren): BadgeChildren => {
```

Comment:
This is kinda neat

Reply:
This is a pretty naive implementation, but meant as a patch since we decided to leave out scaling the badge size with icon sizes for stage 1 and requiring the icon be the default size.

I feel like for development this could benefit from some info logging to inform the consumer of the modification and provide an opportunity for them to "fix" on their own, but left it out since this is stage 1. Also it only covers the case where the children are in a 1 dimensional array.